### PR TITLE
refactor/ts-utils

### PR DESCRIPTION
**Pull request by Dashice**
Converted constants, createRenderBlock, page, actionWithPromise, createEmotionCache to TypeScript.

Added JSDocs comments to functions. Added additional descriptors to common.d.ts

### CODE REVIEW

File: src/utils/createRenderBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    const Component = blocks[name]
+    const children = childrenProp ? childrenProp.map(renderBlock) : undefined
+
+    if (!Component) {
+      console.error(`CIA: 🕵️‍♂️ Block not found: ${name}`)
+      return null
+    }
+
+    return React.createElement(Component, { key: index, renderIndex: index, ...other }, children)
```
Comment (maeertin) [inline src/utils/createRenderBlock.tsx:46]: What's the reason for moving to `React.createElement` here? :)

**Review by alexanderflink — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

File: src/utils/createRenderBlock.tsx (hunk 1 above)
Comment (Dashice) [reply src/utils/createRenderBlock.tsx]: It allowed us to skip defining that a block may have `children?: React.ReactNode` in the `BlockType`. I have reverted to using a JSX approach for more consistency with the rest of the codebase.

**Review by maeertin — APPROVED** [review]
Looking good to me! What about you @alexanderflink? :)

**Comment by maeertin** [comment]
Closing PR as "cia" is now purely a documentation app.

**Pull request closed without merging**